### PR TITLE
test: Add a new unit test for chain.cpp

### PR DESCRIPTION
**Pull request by hunglun**
test: Add a new unit test for chain.cpp

### CODE REVIEW

File: src/test/chain_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#include <chain.h>
+#include <test/util/setup_common.h>
+
+BOOST_AUTO_TEST_SUITE(chain_tests)
+
+BOOST_AUTO_TEST_CASE( get_median_time_past )
+{
+    CBlockIndex blocks[11];
+    blocks[0].nTime=0;
+    for (int i=1; i<11;i++)
```
Comment (jonatack) [inline src/test/chain_tests.cpp:15]: - prefer prefix iterator per doc/developer-notes.md 
```suggestion
    for (int i = 1; i < 11; ++i) {
```

- please run clang-format on your changes

```diff
-BOOST_AUTO_TEST_CASE( get_median_time_past )
+BOOST_AUTO_TEST_CASE(get_median_time_past)
 {
     CBlockIndex blocks[11];
-    blocks[0].nTime=0;
-    for (int i=1; i<11;i++)
-    {
+    blocks[0].nTime = 0;
+    for (int i = 1; i < 11; i++) {
         blocks[i].nTime = i;
-        blocks[i].pprev = &blocks[i-1];
+        blocks[i].pprev = &blocks[i - 1];
     }
     // block nTime already sorted in increasing order
-    BOOST_CHECK( blocks[10].GetMedianTimePast() == blocks[5].nTime);
+    BOOST_CHECK(blocks[10].GetMedianTimePast() == blocks[5].nTime);
 }
```

**Review by jonatack — COMMENTED** [review]
Hi @hunglun, have you looked at the existing tests (i.e. `git grep -ni GetMedianTimePast src/test/`) and the coverage reports at https://marcofalke.github.io/btc_cov to see if there isn't already test coverage on this, and if not, whether it can be added into one of the existing test files? Also, it's best to squash your commits into a final form ready for review (see `CONTRIBUTING.md`). Thanks!

File: src/test/chain_tests.cpp (hunk 1 above)
Comment (hunglun) [reply src/test/chain_tests.cpp:15]: I followed your advice to check the unit test code coverage for chain.h. 
chain.h::ToString() and GetBlockHeader()  are not covered by unit test.

I would like to add new test cases about ToString and GetBlockHeader() to chain_tests.cpp

I also checked existing tests about GetMedianPastTime. Some tests check that the MedianPastTime is within certain bounds; but none of the existing tests is very precise about the expected value of the median past time.

**Comment by jonatack** [comment]
@hunglun if you closed because review seemed slow, please don't be discouraged. It can take time to receive review.

**Pull request closed without merging**